internal/internal_decision_state_machine.go

```diff
@@ -1077,7 +1083,14 @@ func (h *commandsHelper) handleExternalWorkflowExecutionCancelRequested(initiate
 	cancellationID, isExternal := h.scheduledEventIDToCancellationID[initiatedeventID]
 	if !isExternal {
 		command = h.getCommand(makeCommandID(commandTypeChildWorkflow, workflowID))
-		// no state change for child workflow, it is still in CancellationCommandSent
+		asChildWfCmd := command.(*childWorkflowCommandStateMachine)
+		if asChildWfCmd.getState() == commandStateCompletedAfterCancellationCommandSent {
+			// Now we're really done.
+			command.handleCompletionEvent()
+		} else {
+			// We're in the cancellation command sent stage - new state to indicate we have seen the cancel accepted
+			asChildWfCmd.moveState(commandStateCancellationCommandAccepted, eventExternalWorkflowExecutionCancelRequested)
```

Comment:
Can we have handleCancelRequested api on the child workflow state machine?  It does not looks right to call moveState directly from commandsHelper.